5 GET /api/articles/:article_id (comment count)

Chl0e-g opened the pull request:
This pull request contains:
- Refactored previous test to expect the response object to include a comment_count property equal to the total number of comments for the specified article
- Refactored database query in model to return comment_count

Commit:
Refactored previous test and db query to add comment_count property

cjplatten commented on __tests__/app.test.js:
```diff
@@ -58,7 +58,7 @@ describe("/api/articles/:article_id", () => {
         });
     });
-    test("status: 200 - article object in response has these properties: author, title, article_id, body, topic, created_at, votes", () => {
+    test("status: 200 - article object in response has these properties: author, title, article_id, body, topic, created_at, votes, comment_count", () => {
```
This is fine but worth noting that when adding features later that alter response objects it can be a good idea to create a new test rather than amend previous ones to have the property.

cjplatten commented on __tests__/app.test.js:
```diff
@@ -67,6 +67,7 @@ describe("/api/articles/:article_id", () => {
         created_at: expect.any(String),
         votes: 100,
+        comment_count: 11
       };
```
What happens when an article doesn't have any associated comments?

cjplatten reviewed (commented):
See comments

Commit:
Refactored tests and db query to account for zero comments

Chl0e-g commented:
Thanks for the feedback! I've amended the tests to split out the comment_count property into its own test, and added a test for zero comments. I've refactored the model to separate the comment_count db request from the article data db request, so that I still get an article object for an article with zero comments.

Angaca commented on models/articles.models.js:
```diff
@@ -1,20 +1,29 @@
     `
     SELECT * FROM articles
-    WHERE article_id = $1;`,
+    WHERE articles.article_id = $1`,
```
; is missing

Angaca commented on models/articles.models.js:
```diff
@@ -1,20 +1,29 @@
+  const commentCount = db.query(`
+  SELECT COUNT(*)::int AS comment_count 
+  FROM comments
+  WHERE comments.article_id = $1`,
```
; missing here as well

Angaca reviewed (approved):
Good to go!

Commit:
adding semi-colons to SQL queries
